### PR TITLE
doc: fix doxygen INPUT path; brand html output

### LINKED ISSUE #6773: Publish doxygen-generated API pages
We use doxygen to process header files to collect public API information.  Today we use the XML output to feed into the Sphinx/Breathe processing to create the published [API documentation](http://docs.zephyrproject.org/api/api.html).  Doxygen also produces an HTML version of the API documentation that we generate but don't publish.  Some developers have asked for alternative ways of presenting the API docs and cross-references (e.g., #6739).  Publishing the doxygen-generated API material is another.  We should first though, do some customization of the doxygen output to make it more presentable and improve the user experience: Zephyr theme, intro on the doxygen index page, and such.

Also, consider cleaning up the doxygen-generated HTML presentation and completely replace the Sphinx/Breathe-generated API material with that (rather than having duplicated presentations of the same material.)

### PR DESCRIPTION
**Pull request by dbkinder**
Clean up the INPUT paths for doxygen scanning:
- directory scans include sub-directories so no need
  to include them explicitly
- remove excluded (legacy) includes that no longer exist

Sync directory changes in the CMakeLists.txt (as noted in the comments)

Add content to the home page of the doxygen-generated HTML, and add a
Zephyr branding theme.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

Fixes: #6773

### CODE REVIEW

**Comment by dbkinder** [comment]
This will give us a [doxygen-generated home page](https://docs.zephyrproject.org/apidoc/latest/index.html) generated by the CI nightly docs build, that looks like this:

Screenshot from 2019-06-25 16-15-18

**Review by marc-hb — COMMENTED** [review]
It's difficult to come with a good commit subject line that fits in +-75 characters because you have several unrelated changes in one commit. However I think it's still possible to find something with more information than "improvements". Giving it a try: "doxygen: fix INPUT path; add html output" ?

**Review by nashif — APPROVED** [review]
LGTM

**Comment by marc-hb** [comment]
BTW I'm working on a fix for the codeowers check.

**Comment by marc-hb** [comment]
@nashif fixed these in commit 549bd60624ed96

```
Codeowners issues
Path '/include/drivers/modem/' not found, in CODEOWNERS
Path '/include/drivers/ioapic.h' not found, in CODEOWNERS
Path '/include/drivers/loapic.h' not found, in CODEOWNERS
Path '/include/drivers/serial/uart_ns16550.h' not found, in CODEOWNERS
```
Permanent codeowners logic fix submitted in PR #17087

File: doc/custom-doxygen/header.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<script type="text/javascript" src="$relpath^jquery.js"></script>
+<script type="text/javascript" src="$relpath^dynsections.js"></script>
+$treeview
+$search
+$mathjax
+<link href="$relpath^$stylesheet" rel="stylesheet" type="text/css" />
+$extrastylesheet
+</head>
+<body>
+<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply doc/custom-doxygen/header.html]: This is the header that came from ``doxygen -w html header.html footer.html customdoxygen.css`` with a slight tweak to make the logo a clickable link to the docs home page.

**Comment by dbkinder** [comment]
retest

**Pull request merged**